### PR TITLE
Add a data attribute to add additional runfiles to the appimage

### PR DESCRIPTION
**Pull request by woutervdstoel**
This allows adding custom data to an appimage. The use case is a setup in which one would normally use a `label_flag` option to add additional runfiles to a given executable. We want to encode some useful values of that `label_flag` in the appimage.

### CODE REVIEW

File: tests/test.py [hunk 1]
```diff
@@ -17,6 +17,14 @@ def test_datadep() -> None:
     assert data_dep.is_file(), f"{data_dep} does not exist"
     assert (size := data_dep.stat().st_size) == 591, f"{data_dep} has wrong size {size}"
 
+    data_dep = Path("tests/appimage_data_file.txt")
```
Comment (lalten) [inline tests/test.py:20]: I think it would be nice to put these in a separate test to make clear that this is not the data dep of the binary, but of the appimage. Make sure to call it in the list of tests at the bottom of this file.

File: tests/test.py [hunk 2]
```diff
@@ -17,6 +17,14 @@ def test_datadep() -> None:
...
+    assert data_dep.is_file(), f"{data_dep} does not exist"
+    assert (size := data_dep.stat().st_size) == 13, f"{data_dep} has wrong size {size}"
```
Comment (lalten) [inline tests/test.py:22]: That's a little obscure, tbh I am not sure anymore why I tested the size of these files. Probably to rule out that they're symlinks, and the right file?

Comment (woutervdstoel) [reply tests/test.py:22]: Yeah I guess it's important to make sure that they're not symlinks into the source/bazel cache. Was that ever a problem?

File: appimage/private/runfiles.bzl [hunk 3]
```diff
@@ -87,15 +87,16 @@ def collect_runfiles_info(ctx):
     Returns:
         struct with infos about files needed by app.
     """
-    runfiles_list = _default_runfiles(ctx.attr.binary).to_list()
-    file_map = {_final_file_path(ctx, f): f for f in runfiles_list}
+    runfiles_list = depset(ctx.files.data, transitive = [_default_runfiles(ctx.attr.binary)] + [_default_runfiles(d) for d in ctx.attr.data]).to_list()
+    symlinks_list = depset(transitive = [_default_symlinks(ctx.attr.binary)] + [_default_symlinks(d) for d in ctx.attr.data]).to_list()
+    emptyfiles_list = depset(transitive = [_default_emptyfiles(ctx.attr.binary)] + [_default_emptyfiles(d) for d in ctx.attr.data]).to_list()
```
Comment (lalten) [inline appimage/private/runfiles.bzl:92]: So the depset is just for de-duplication at least for symlinks and emptyfiles? Might be nice to add an explaining comment.

File: tests/test.py (hunk 2 above)
Comment (lalten) [reply tests/test.py:25]: Probably yes. In a previous code base I worked on there were so many edge cases, and I hit them all :D
